### PR TITLE
HAPP - 412 Filter health records via type

### PR DESCRIPTION
**Pull request by rashmi-bambhania**

### CODE REVIEW

File: app/src/main/java/ca/bc/gov/bchealth/ui/login/BcscAuthFragment.kt [hunk 1]
```diff
@@ -172,6 +177,8 @@ class BcscAuthFragment : Fragment(R.layout.fragment_bcsc_auth) {
             viewModel.resetAuthStatus()
             when (authStatus.tosAccepted) {
                 TOSAccepted.ACCEPTED -> {
+                    // clear timeline filter
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/ui/login/BcscAuthFragment.kt:180]: Why we are resetting filters in the BCSC auth ViewModel

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/filter/FilterViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package ca.bc.gov.bchealth.ui.healthrecord.filter
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class FilterViewModel @Inject constructor() : ViewModel() {
+    var timelineTypeFilter: List<TimelineTypeFilter> = listOf(TimelineTypeFilter.ALL)
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/filter/FilterViewModel.kt:9]: convert it to flow

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/filter/FilterFragment.kt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                    R.id.chip_immunizations -> {
+                        filterList.add(TimelineTypeFilter.IMMUNIZATION)
+                    }
+                }
+            }
+            if (filterList.isNullOrEmpty()) {
+                filterList.add(TimelineTypeFilter.NONE)
+            }
+            filterSharedViewModel.timelineTypeFilter = filterList
+            parentFragmentManager.setFragmentResult(
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/filter/FilterFragment.kt:99]: remove back stack result passing

**Review by pinakinkansara — CHANGES_REQUESTED** [review]

File: app/src/main/java/ca/bc/gov/bchealth/ui/login/BcscAuthFragment.kt (hunk 1 above)
Comment (rashmi-bambhania) [reply app/src/main/java/ca/bc/gov/bchealth/ui/login/BcscAuthFragment.kt:180]: Removed from here

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/filter/FilterFragment.kt (hunk 3 above)
Comment (rashmi-bambhania) [reply app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/filter/FilterFragment.kt:99]: Done

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/filter/FilterViewModel.kt (hunk 2 above)
Comment (rashmi-bambhania) [reply app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/filter/FilterViewModel.kt:9]: Done

**Pull request merged**